driver/csiplugin/controllerserver.go

```diff
@@ -707,6 +715,79 @@ func checkSCSupportedParams(params map[string]string) (string, bool) {
 	return strings.Join(invalidParams[:], ", "), false
 }
 
+
+// validateVACParams is used to check whether valid AFM tuning parameters are passed or not
+// As part of initial implementation only 6 parameters are considered for tuning by default
+// afmObjectSyncOpenFiles,afmNumFlushThreads,afmPrefetchThreshold,afmObjectFastReaddir,afmFileOpenRefreshInterval,afmNumReadThreads (Default parameters)
+// Values to the parameters are configured through VAC (volume attributes class). If not then default values are considered for tuning
+func validateVACParams(ctx context.Context, mutableParams map[string]string) map[string]interface{}{
+	loggerId := utils.GetLoggerId(ctx)
+	afmTuningParams := make(map[string]interface{})
+
+	for vacKey,vacValue := range mutableParams{
+	 switch vacKey{
+
+         case connectors.AfmReadSparseThreshold:
+		afmReadSparseThresholdValue,_ := strconv.Atoi(vacValue)
+		if afmReadSparseThresholdValue < 0 && afmReadSparseThresholdValue > 4294967296{
```

Comment:
What if we return the error here instead of using default value ?

Reply:
Done